app/views/data_view.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+from app.models import Donor, Donation, Item
+from django.http import HttpResponse, JsonResponse, HttpResponseBadRequest
+from django.views import View
+import simplejson as json
+
+@login_required(login_url='/login')
+def aggregate_value(request):
+    '''if request.GET['interval'] is false, return value of all items
+    otherwise, return array of values for given interval
+    '''
+    try:
+        start_date = request.GET['start_date']
+        end_date = request.GET['end_date']
+
+        items = __getModelsGivenInterval('item', start_date, end_date)
+        
+        item_date_pairs = map(__generatePairs, items)
```

Comment:
After you map you need to reduce the pairs with same dates and add them up into a single pair.